samcli/local/apigw/local_apigw_service.py

```diff
@@ -397,6 +403,78 @@ def _parse_lambda_output(lambda_output, binary_types, flask_request):
 
         return status_code, headers, body
 
+    @staticmethod
+    def _parse_v2_payload_format_lambda_output(lambda_output, binary_types, flask_request):
+        """
+        Parses the output from the Lambda Container
+
+        :param str lambda_output: Output from Lambda Invoke
+        :return: Tuple(int, dict, str, bool)
+        """
+        # pylint: disable-msg=too-many-statements
+        try:
+            json_output = json.loads(lambda_output)
+        except ValueError as ex:
+            raise LambdaResponseParseException("Lambda response must be valid json") from ex
+
+        # lambda can return any valid json response in payload format version 2.0.
+        # response can be a simple type like string, or integer
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.response
+        if isinstance(json_output, dict):
+            body = json_output.get("body") if "statusCode" in json_output else json.dumps(json_output)
+        else:
+            body = json_output
+            json_output = {}
+
+        if not body:
+            LOG.warning("Lambda returned empty body!")
+
+        status_code = json_output.get("statusCode") or 200
+        headers = LocalApigwService._merge_response_headers(
+            json_output.get("headers") or {}, json_output.get("multiValueHeaders") or {}
+        )
```

Comment:
These values are not required to be provided by the Lambda function's response.

Reply:
these values can be provided by Lambda, but in case if they are not provided, we should return some default values instead.
https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.response